lib/hyacinth/adapters/external_identifier_adapter/memory.rb

```diff
@@ -0,0 +1,67 @@
+module Hyacinth
+  module Adapters
+    module ExternalIdentifierAdapter
+      class Memory < Abstract
+        attr_reader :identifiers
+
+        def initialize(adapter_config = {})
+          super(adapter_config)
+          @identifiers = {}
+        end
+
+        # @param id [String]
+        # @return [Boolean] true if this adapter can handle this type of identifier
+        def handles?(_id)
+          true
+        end
+
+        # Generates a new persistent id, ensuring that nothing currently uses that identifier.
+        # @return [String] a new id
+        def mint
+          new_id = Random.rand.to_s
+          @identifiers[new_id] = {}
```

Comment:
Since this method is supposed to ensure that nothing currently uses the minted identifier, should you be checking for the presence of new_id in `@identifiers` and looping until you don't find it?

Reply:
We can, sure - tho (as the in-memory implementation for testing) I'm inclined to just trust the random number generator?

Reply:
I know the odds are really low, but since it's pretty easy to do, do you mind if we add this?  It'll make me feel slightly better.

```
def mint
  loop do 
    new_id = Random.rand.to_s
    break unless @identifiers.key?(new_id)
  end

  @identifiers[new_id] = {}
  new_id
end
```